recipes/bzip2/all/test_package/conanfile.py

```diff
@@ -1,17 +1,25 @@
 import os
-from conans import ConanFile, CMake, tools
+from conans import ConanFile, tools
+from conan.tools.cmake import CMake
+from conan.tools.layout import cmake_layout
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "arch", "build_type"
-    generators = "cmake", "cmake_find_package"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
 
     def build(self):
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
+    def configure(self):
+        del self.settings.compiler.libcxx
+
+    def layout(self):
+        cmake_layout(self)
+
     def test(self):
         if not tools.cross_building(self.settings):
-            bin_path = os.path.join("bin", "test_package")
-            self.run("%s --help" % bin_path, run_environment=True)
+            cmd = os.path.join(self.cpp.build.bindirs[0], "test_package")
```

Comment:
I don't know if it's safe, e.g. if bindirs is an empty

Reply:
It is a bindir of the current package. As you are declaring the layout(), you know there is always a bindir. Actually were always a bindir unless you remove it from the cppinfo.